Helper/Webhook.php

```diff
@@ -0,0 +1,1883 @@
+     */
+    private function getTransitionState(Notification $notification, $currentOrderState): string
+    {
+        $webhookNotificationItem = \Adyen\Webhook\Notification::createItem([
+            'eventCode' => $notification->getEventCode(),
+            'success' => $notification->getSuccess(),
+            'additionalData' => !empty($notification->getAdditionalData())
+                ? $this->serializer->unserialize($notification->getAdditionalData()) : null,
+        ]);
+        $processor = ProcessorFactory::create($webhookNotificationItem, $currentOrderState, $this->logger);
+
+        return $processor->process();
+    }
+
+    /**
+     * @param Notification $notification
+     * @param $transitionState
+     * @throws Exception
+     * @throws LocalizedException
+     */
+    private function handleOrderTransition(Notification $notification, $transitionState): void
+    {
+        $previousAdyenEventCode = $this->order->getData('adyen_notification_event_code');
+        switch ($transitionState) {
+            case PaymentStates::STATE_PAID:
+                if (Notification::CAPTURE == $notification->getEventCode()) {
+                    /*
+                     * ignore capture if you are on auto capture
+                     * this could be called if manual review is enabled and you have a capture delay
+                     */
+                    if (!$this->isAutoCapture($notification->getPaymentMethod())) {
+                        $this->handleManualCapture($notification);
+                    }
+                } elseif (in_array($notification->getEventCode(), [Notification::REFUND, Notification::REFUND_FAILED])) {
+                    // Webhook module returns PAID for failed refunds, trigger admin notice
+                    $this->addRefundFailedNotice($notification);
+                } else {
+                    $this->authorizePayment($notification);
+                }
+                break;
+            case PaymentStates::STATE_FAILED:
+            case PaymentStates::STATE_CANCELLED:
+                $this->cancelPayment($notification, $previousAdyenEventCode);
+                break;
+            case PaymentStates::STATE_REFUNDED:
+                $this->refundPayment($notification);
+                break;
+        }
+    }
+
+    private function handleUnchangedStates(Order $order, Notification $notification): void
+    {
+        switch ($notification->getEventCode()) {
+            case Notification::PENDING:
+                if ($this->configHelper->getConfigData(
+                    'send_email_bank_sepa_on_pending',
+                    'adyen_abstract',
+                    $this->order->getStoreId()
+                )
+                ) {
+                    // Check if payment is banktransfer or sepa if true then send out order confirmation email
+                    if ($this->isBankTransfer($notification->getPaymentMethod()) || $notification->getPaymentMethod() == 'sepadirectdebit') {
```

Comment:
Does this mean that SEPA confirmation emails will only be sent when the order is in PENDING and we receive a webhook with PENDING event code?

Reply:
Not **only** if, confirmation email is sent in the `authorizePayment` method as well, this case only makes sure we also send the emails for bank transfer and sepadirect even if the notification event code is PENDING.
BTW this only checks for the Notification event code not the order state, and only if the state remains unchanged from the webhook module